src/summary.cpp

```diff
@@ -0,0 +1,142 @@
+#include <lo2s/config.hpp>
+#include <lo2s/summary.hpp>
+#include <lo2s/util.hpp>
+
+#include <boost/filesystem.hpp>
+
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+#include <ratio>
+#include <array>
+
+extern "C" {
+#include <sys/time.h>
+}
+
+namespace lo2s
+{
+
+Summary& summary()
+{
+    static Summary s;
+    return s;
+}
+
+Summary::Summary()
+: start_wall_time_(std::chrono::steady_clock::now()), num_wakeups_(0),
+  thread_count_(0), trace_dir_(""), exit_code_(0)
```

Comment:
`trace_dir_("")` is equivalent to the default constructor of std::string, so you can omit it. Less code is better code.